libs/wingsdk/src/target-sim/resource.inflight.ts

```diff
@@ -136,9 +136,23 @@ export class Resource implements IResourceClient, ISimulatorResourceInstance {
 
   public async save(): Promise<void> {}
 
-  public async plan(): Promise<UpdatePlan> {
-    // TODO: support other update plans
-    return UpdatePlan.REPLACE;
+  public async plan(invalidated: boolean): Promise<UpdatePlan> {
+    // TODO: support customizing update plans
+    if (invalidated) {
+      return UpdatePlan.REPLACE;
+    }
+
+    // Check if any of the bundled files have changed since the last bundling
+    const bundleInvalidated = await isBundleInvalidated(
+      this.resolvedSourceCodeFile,
+      this.bundle!,
```

Comment:
Maybe as a future refactor, would it make sense to encapsulate this behavior into `Bundle`?

And then it would be something like:

```js
if (bundle.isInvalidated()) ...
```

Or something like this...

Could be nice to also encapsulate `ensureBundled` into this class so there's less duplication across consumers.

Reply:
Yeah, I agree it it makes sense to make Bundle into its own stateful class etc. I'll open an issue